datagateway_api/src/search_api/panosc_mappings.py

```diff
@@ -26,6 +26,49 @@ def __init__(
                     f"An error occurred while trying to load the PaNOSC mappings: {e}",
                 )
 
+    def get_icat_mapping(self, panosc_entity_name, field_name):
+        """
+        This function searches the PaNOSC mappings and retrieves the ICAT translation
+        from the PaNOSC input. Fields in the same entity can be found, as well as fields
+        from related entities (e.g. Dataset.files.path) via recursion inside this
+        function.
+
+        :param panosc_entity_name: A PaNOSC entity name e.g. "Dataset"
+        :type panosc_entity_name: :class:`str`
+        :param field_name: PaNOSC field name to fetch the ICAT version of e.g. "name"
+        :type field_name: :class:`str`
+        :return: Tuple containing the PaNOSC entity name (which will change from the
+            input if a related entity is found) and the ICAT field name
+            mapping/translation from the PaNOSC data model
+        :raises FilterError: If a valid mapping cannot be found
+        """
+
+        log.info(
+            "Searching mapping file to find ICAT translation for %s",
+            f"{panosc_entity_name}.{field_name}",
+        )
+
+        try:
+            icat_mapping = mappings.mappings[panosc_entity_name][field_name]
```

Comment:
This should be like the below otherwise the method needs to be static
```suggestion
            icat_mapping = self.mappings[panosc_entity_name][field_name]
```